flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosResourceManager.java

```diff
@@ -201,7 +201,6 @@ public MesosResourceManager(
 		final ContaineredTaskManagerParameters containeredTaskManagerParameters = taskManagerParameters.containeredParameters();
 		this.slotsPerWorker = updateTaskManagerConfigAndCreateWorkerSlotProfiles(
 			flinkConfig, containeredTaskManagerParameters.taskManagerTotalMemoryMB(), containeredTaskManagerParameters.numSlots());
-		setFailUnfulfillableRequest(true);
```

Comment:
This is a good finding. Don't we have tests for Mesos which test this behavior? Given the previous code, unfulfillable requests should have failed right away, right?

Reply:
Yes, unfulfillable requests should have failed right away for Yarn and Mesos. 

We have `SlotManagerFailUnfulfillableTest` that tests this behavior at the scope of slot manager, but no tests at the scope of resource manager for Yarn and Mesos. 

I've update the PR changing the default value of `SlotManager#failUnfulfillableRequest` to true and set it to false on start of `StandaloneResourceManager`. So the Yarn / Mesos resource manager do not need to do anything special regarding failing unfulfillable requests. I don't think we need more test cases other than `SlotManagerFailUnfulfillableTest` to test this behavior for Yarn / Mesos.